cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -39,6 +39,14 @@
 (rule (simplify (imul ty (ineg ty x) (ineg ty y)))
       (subsume (imul ty x y)))
 
+;; iabs(ineg(x)) == iabs(x).
+(rule (simplify (iabs ty (ineg ty x)))
+      (subsume (iabs ty x)))
```

Comment:
There have been some discussions about `subsume` and how to use it well. We're still not sure, because it's complicated.

For example, here you aren't just claiming that `iabs(x)` is better than `iabs(ineg(x))`, you're also claiming that it's better than every other rule that matches here. Maybe that's true, but I don't know how to convince myself either way.

Using `subsume` never causes the optimizer to produce incorrect results, so using it here isn't wrong. It just cuts off the search for other results which might have been better.

My current opinion is that we should only use `subsume` when rewriting to a constant (where it's easy to prove that there is no better alternative) and _maybe_ when rewriting to a term which came directly from the pattern on the left-hand side of the rule, like you did below with iabs/iabs. I've had trouble convincing myself of even that though.

I missed reviewing your PRs like the `-x*-y` rewrite immediately above this, and that's okay. I'm not going to ask you to remove the subsumes from those; it's not that big a deal. But would you remove this one?